script_vts.py

```diff
@@ -207,78 +183,61 @@ def formating_file():
     else:
-        response = search_api_request(api_key, args.string)
-
-        if response is None:  # If we receive nothing quit with error
-            exit(1)
-
-        if parse_json(response, args.string, file_reputation, filename) is True:  # check if we need to call write_report() for file reputation
-            response_file_reputation = file_reputation_api_request(api_key, args.string)
-            append_report(filename, "[+] Response from File reputation request: \n")
-            append_report(filename, response_file_reputation)
-
-    print(
-        f"[+] The file report can be found as {filename}.txt in {Path.cwd()}")
-
-    exit(0)
+        report_data_dic = handle_file_report(report_data_dic, args.string)
+    for r in report_data_dic:
+        append_report(filename, r)
+    print(f"[+] The file report can be found as {filename}.txt in {Path.cwd()}")
+    return 0
+
+
+def handle_file_report(report_data_dic, string_argument):
+    """Function used to handle our file report"""
+    response = search_api_request(api_key, string_argument)
+    if response is None:  # If we receive nothing quit with error
+        print("[-] We didn't receive something from our search request.")
+        return 1
+    error, report_data_dic, string_parse_json, res_dic, analysis_stat, res_parse_json = parse_json(report_data_dic, response, string_argument)
+    if error is not True:
+        data_process_variable,  report_data_dic = data_processing(string_parse_json, res_dic, analysis_stat, res_parse_json, report_data_dic)
+        if data_process_variable is True:  # check if we need to call for file reputation
+            response_file_reputation = file_reputation_api_request(api_key, string_argument)
+            report_data_dic.append("[+] Response from File reputation request: \n")
+            report_data_dic.append(response_file_reputation)
+    return report_data_dic
 
 
 if __name__ == '__main__':
-
-    if checking_arguments(sys.argv) is None:  # Check if an argument is provided
+    if verify_at_least_one_command_line_argument(sys.argv) is None:  # Check if an argument is provided
         print("[-] Please provide at least one command line argument")
         exit(1)
 
-    parser = argparse.ArgumentParser()  # Create the parser to have arguments in the script
-
-    parser.add_argument('--string', type=str, )  # Adding string argument
-
-    parser.add_argument('--file', type=argparse.FileType('r'))  # Adding File argument
-
-    args = parser.parse_args()  # Parse the argument
+    args = parse_arguments()
 
     file_arg = False  # Set file argument to false until we know there's one
 
     if args.file:  # Set to true if we find that it is used as cmdline argument, Remove any blank line so we can send proper string to the api_req()
         file_arg = True
-
         formating_file()
 
     string = ""
 
     path = get_path(string)  # Call the get_path function to have the path of the API Key File
 
```

Comment:
Remove this line, the above and below lines are related.

Reply:
Hum, weird this line is actually removed there's no space between `file_arg = True` and `formating_file`. Is it good like this or do you want me to put a space between the two instructions?